src/forms/useGroupForm.ts

```diff
@@ -0,0 +1,115 @@
+import { IGroupForm } from '@/interfaces/formInterfaces';
+
+import { useForm } from 'react-hook-form';
+
+import { useEffect, useMemo, useState } from 'react';
+import { useAppSelector } from '@/hooks/common';
+import { selectMasterClasses, selectClasses, selectSubClasses } from '@/reducers/classSlice';
+import { selectDistricts, selectDivisions, selectSubDivisions } from '@/reducers/locationSlice';
+import useClassOptions from '@/hooks/useClassOptions';
+import useLocationOptions from '@/hooks/useLocationOptions';
+
+interface ISelectionState {
+  selectedClass: string | undefined;
+  selectedLocation: string | undefined;
+}
+const useGroupValues = () => {
+  const formValues = useMemo(
+    () => ({
+      name: '',
+      masterClass: {},
+      class: {},
+      subClass: {},
+      district: {},
+      division: {},
+      subDivision: {},
+      projectsForSubmit: [],
+    }),
+    [],
+  );
+  return { formValues };
+};
+
+const useGroupForm = () => {
+
+
+  const [selections, setSelections] = useState<ISelectionState>({
+    selectedClass: '',
+    selectedLocation: '',
+  });
+  const { selectedClass, selectedLocation } = selections;
+  const { masterClasses, classes, subClasses } = useClassOptions(selectedClass);
+  const { districts, divisions, subDivisions } = useLocationOptions(selectedLocation);
+  const { formValues } = useGroupValues();
+
+  const formMethods = useForm<IGroupForm>({
+    defaultValues: useMemo(() => formValues, [formValues]),
+    mode: 'onSubmit',
+  });
+
+
+  const { watch, setValue } = formMethods;
+  useEffect(() => {
+    const subscription = watch((value, { name }) => {
+      switch (name) {
+        case 'masterClass':
```

Comment:
These cases could be refactored as fall-through's (if-conditions could be wrapped with curly braces if it looks more readable),

```
case 'masterClass':
case 'class':
case 'subClass':
  setSelections((current) => ({ ...current, selectedClass: value[name].value }));
  
  if (name === 'masterClass') setValue('class', { label: '', value: '' });
  if (name === 'masterClass' || name === 'class') setValue('subClass', { label: '', value: '' });

case 'district':
case 'division':
case 'subDivision':
  setSelections((current) => ({ ...current, selectedLocation: value[name].value }));
  
  if (name === 'district') setValue('division', { label: '', value: '' });
  if (name === 'district' || name === 'division') setValue('subDivision', { label: '', value: '' });
```